### ISSUE TITLE
Bug when swapping axes along the second dimension of the `points` argument in `Raster._transform`.

### ISSUE THREAD

**Issue by yszhang95** [yszhang95/tred#5]
The `tdim` defined in the [source](https://github.com/brettviren/tred/blob/c1e60a5b5df0ef5f8975bacdd6960ef64c5ff3d4/src/tred/graph.py#L111) is redundant and cause bugs when it is not set to -1.

Existing code puts axes for pixel (old axes) to first few dimensions, according to the [line](https://github.com/brettviren/tred/blob/c1e60a5b5df0ef5f8975bacdd6960ef64c5ff3d4/src/tred/graph.py#L133).
The assignment may override the axis tdim (new axes), suggested in the [line](https://github.com/brettviren/tred/blob/c1e60a5b5df0ef5f8975bacdd6960ef64c5ff3d4/src/tred/graph.py#L139).

It is easy to show by the following:
```
import torch
import tred.graph as tg
r = tg.Raster(1.6, (1, 1, 1), pdims=(1,2), tdim=0)
tail = torch.tensor([[3,4,5]])
time = torch.tensor([2])
print(r._transform(tail, time)) # we get (2,5,0) while we expect (4, 5, 2).
```

Proposed fix while keeping interfaces looks like the following pseudo-code:
```
pdims = (1, 2)
tdim = 1
vdim = len(pdims) + 1
old_tdim = (set(range(vdim)) - set(pdims)).pop()
points = torch.tensor([[3,4,5]])
axes = list(iter(pdims))
axes.insert(tdim, old_tdim)
points = points[:, axes]
print(points) # we get [4, 3, 5]
points[:, tdim] = time
print(points) # we get [4, 2, 5]
```

**Comment by yszhang95** [comment]
Fixed in commit 1518d77.

**Comment by yszhang95** [comment]
The proposed solution does not work when `tdim` <0.

A fix is given in 6b5d231c69972fd0030c06e29c4108d296f69bdb.

**Comment by yszhang95** [comment]
Merged in #18.
